PWGDQ/Core/VarManager.h

```diff
@@ -2939,10 +2956,27 @@ void VarManager::FillPairME(T1 const& t1, T2 const& t2, float* values)
 
   // TODO: provide different computations for vn
   // Compute the scalar product UQ for two muon from different event using Q-vector from A, for second and third harmonic
-  values[kU2Q2Ev1] = values[kQ2X0A1] * std::cos(2 * v1.Phi()) + values[kQ2Y0A1] * std::sin(2 * v1.Phi());
-  values[kU2Q2Ev2] = values[kQ2X0A2] * std::cos(2 * v2.Phi()) + values[kQ2Y0A2] * std::sin(2 * v2.Phi());
-  values[kCos2DeltaPhiMu1] = std::cos(2 * (v1.Phi() - v12.Phi()));
-  values[kCos2DeltaPhiMu2] = std::cos(2 * (v2.Phi() - v12.Phi()));
+  float Psi2A1 = getEventPlane(2, values[kQ2X0A1], values[kQ2Y0A1]);
+  float Psi2A2 = getEventPlane(2, values[kQ2X0A2], values[kQ2Y0A2]);
+  values[kCos2DeltaPhiEv1] = TMath::Cos(2 * (v1.Phi() - Psi2A1));
+  values[kCos2DeltaPhiEv2] = TMath::Cos(2 * (v2.Phi() - Psi2A2));
+  values[kU2Q2Ev1] = values[kQ2X0A1] * TMath::Cos(2 * v1.Phi()) + values[kQ2Y0A1] * TMath::Sin(2 * v1.Phi());
+  values[kU2Q2Ev2] = values[kQ2X0A2] * TMath::Cos(2 * v2.Phi()) + values[kQ2Y0A2] * TMath::Sin(2 * v2.Phi());
+
+  values[kCos2DeltaPhiMu1] = TMath::Cos(2 * (v1.Phi() - v12.Phi()));
+  values[kCos2DeltaPhiMu2] = TMath::Cos(2 * (v2.Phi() - v12.Phi()));
+
+  values[kV2SP1] = values[kU2Q2Ev1] / values[kTwoR2SP1];
+  values[kV2SP2] = values[kU2Q2Ev2] / values[kTwoR2SP2];
+  values[kV2EP1] = values[kCos2DeltaPhiEv1] / values[kTwoR2EP1];
+  values[kV2EP2] = values[kCos2DeltaPhiEv2] / values[kTwoR2EP2];
+
+  float V2ME_SP = values[kV2SP1] * values[kCos2DeltaPhiMu1] + values[kV2SP2] * values[kCos2DeltaPhiMu2];
+  float V2ME_EP = values[kV2EP1] * values[kCos2DeltaPhiMu1] + values[kV2EP2] * values[kCos2DeltaPhiMu2];
+  values[kV2ME_SP] = std::isnan(V2ME_SP) || std::isinf(V2ME_SP) ? 0. : V2ME_SP;
+  values[kWV2ME_SP] = std::isnan(V2ME_SP) || std::isinf(V2ME_SP) ? 0. : 1.0;
+  values[kV2ME_EP] = std::isnan(V2ME_EP) || std::isinf(V2ME_EP) ? 0. : V2ME_EP;
+  values[kWV2ME_EP] = std::isnan(V2ME_EP) || std::isinf(V2ME_EP) ? 0. : 1.0;
```

Comment:
All these calculations have to be done only if they are requested, otherwise they will be run by all the analyses which will slow down analyses. So please add a protection so they are computed only in the case they are needed.

Reply:
Thanks a lot for your comment. So then I will need to revert my last commit; I removed it because it touched also part of PWGEM code where they used also table-reader

Reply:
I reverted my last commit, and changed the label as well to pass the `Pull Request Labeler` check